ctru-rs/examples/ir-user.rs

```diff
@@ -0,0 +1,251 @@
+
+        // Check if we should start the connection
+        if hid.keys_down().contains(KeyPad::KEY_A) && !is_connected {
+            println!("Attempting to connect to the CPP");
+
+            match demo.connect_to_cpp(&hid) {
+                ConnectionResult::Connected => is_connected = true,
+                ConnectionResult::Canceled => break,
+            }
+        }
+
+        gfx.flush_buffers();
+        gfx.swap_buffers();
+        gfx.wait_for_vblank();
+    }
+}
+
+struct CirclePadProDemo<'screen> {
+    top_console: Console<'screen>,
+    bottom_console: Console<'screen>,
+    ir_user: IrUser,
+    connection_status_event: Handle,
+    receive_packet_event: Handle,
+}
+
+enum ConnectionResult {
+    Connected,
+    Canceled,
+}
+
+impl<'screen> CirclePadProDemo<'screen> {
+    fn new(top_console: Console<'screen>, bottom_console: Console<'screen>) -> Self {
+        bottom_console.select();
+        println!("Welcome to the ir:USER / Circle Pad Pro Demo");
+
+        println!("Starting up ir:USER service");
+        let ir_user = IrUser::init(
+            PACKET_BUFFER_SIZE,
+            PACKET_COUNT,
+            PACKET_BUFFER_SIZE,
+            PACKET_COUNT,
+        )
+        .expect("Couldn't initialize ir:USER service");
+        println!("ir:USER service initialized");
+
+        // Get event handles
+        let connection_status_event = ir_user
+            .get_connection_status_event()
+            .expect("Couldn't get ir:USER connection status event");
+        let receive_packet_event = ir_user
+            .get_recv_event()
+            .expect("Couldn't get ir:USER recv event");
+
+        Self {
+            top_console,
+            bottom_console,
+            ir_user,
+            connection_status_event,
+            receive_packet_event,
+        }
+    }
+
+    fn print_status_info(&self) {
+        self.top_console.select();
+        self.top_console.clear();
+        println!("{:#x?}", self.ir_user.get_status_info());
+        self.bottom_console.select();
+    }
+
+    fn connect_to_cpp(&self, hid: &Hid) -> ConnectionResult {
+        // Connection loop
+        loop {
+            hid.scan_input();
+            if hid.keys_held().contains(KeyPad::KEY_START) {
+                return ConnectionResult::Canceled;
+            }
+
+            // Start the connection process
+            self.ir_user
+                .require_connection(IrDeviceId::CirclePadPro)
+                .expect("Couldn't initialize circle pad pro connection");
+
+            // Wait for the connection to establish
+            if let Err(e) = self
+                .connection_status_event
+                .wait_for_event(Duration::from_millis(100))
+            {
+                if !e.is_timeout() {
+                    panic!("Couldn't initialize circle pad pro connection: {e}");
+                }
+            }
+
+            self.print_status_info();
+            if self.ir_user.get_status_info().connection_status == 2 {
```

Comment:
could `2` be defined (in `ir_user`, I suppose) as a `#[repr(u8)] enum Status` or something like that?